util/util.go

```diff
@@ -99,23 +104,26 @@ func LdSize(d ...[]byte) uint64 {
 	return sum + uint64(n)
 }
 
-func LdRead(r *bufio.Reader) ([]byte, error) {
+func LdRead(r *bufio.Reader) ([]byte, uint64, error) {
 	if _, err := r.Peek(1); err != nil { // no more blocks, likely clean io.EOF
-		return nil, err
+		return nil, 0, err
```

Comment:
This whole check seems unnecessary because you should get the same error when calling `binary.ReadUvarint(r)` below